### PR TITLE
feat: set ratelimit values by route in yaml files

### PR DESCRIPTION
**Pull request by jerlio**
Allow using this bundle with configuration in yaml files like this:

```yaml
bedrock_rate_limit:
    limit: 1000
    period: 60

    routes:
        get_foobar:
            limit: 500
            period: 10
        post_foobar:
            period: 10
```

With that we'll be able to set different values on rate limit by route and by env.

## TODO
* [x] ~Add listener and configuration~
* [x] ~Unit tests~
* [x] ~Add config example on readme~

### CODE REVIEW

**Review by titoko — COMMENTED** [review]
Looks good for me

**Comment by nalscher** [comment]
@jerlio The only thing i see, it's that we must find a way to desactivate annotation process if routes are configured edit because we can be rateLimit by global conf before by route and it's not really what we want

**Review by maximebilliau — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -101,3 +104,21 @@ This annotation requires a list of endpoints and accepts parameters to customize
 * )
 */
 ```
+
+#### In YAML
+
+You also may add your rate limits in configuration files (Yaml) with the route name. If `period` or `limit` is not
+defined for a route, the bundle will take the common option.
```
Comment (Anicky) [inline README.md:111]:
```suggestion
You may also add your rate limits in configuration files (Yaml) with the route name. If `period` or `limit` is not defined for a route, the bundle will take the common option.
```

**Review by Melfo01 — APPROVED** [review]
LGTM

File: tests/EventListener/ReadRateLimitConfigurationListenerTest.php [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ControllerEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+class ReadRateLimitConfigurationListenerTest extends TestCase
+{
+    private ReadRateLimitConfigurationListener $readRateLimitConfigurationListener;
+
+    /** @var array<RateLimitModifierInterface|MockObject> */
+    private $rateLimitModifiers;
```
Comment (Anicky) [inline tests/EventListener/ReadRateLimitConfigurationListenerTest.php:22]:
```suggestion
    private array $rateLimitModifiers;
```

File: src/EventListener/ReadRateLimitConfigurationListener.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+use Bedrock\Bundle\RateLimitBundle\Model\RateLimit;
+use Bedrock\Bundle\RateLimitBundle\RateLimitModifier\RateLimitModifierInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ControllerEvent;
+
+class ReadRateLimitConfigurationListener implements EventSubscriberInterface
+{
+    /** @var iterable<RateLimitModifierInterface> */
+    private $rateLimitModifiers;
```
Comment (Anicky) [inline src/EventListener/ReadRateLimitConfigurationListener.php:15]:
```suggestion
    private iterable $rateLimitModifiers;
```

**Review by Anicky — APPROVED** [review]

**Review by nalscher — APPROVED** [review]

**Pull request merged**